devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -773,7 +773,7 @@ def zwaveEvent(UserCodeReport cmd) {
 				result << codeSetEvent(lockCodes, codeID, codeName)
 			} else {
 				map.value = "$codeID changed"
-				map.isStateChange = false
+				map.isStateChange = true
```

Comment:
While scanning let's say 8 slots are found to be filled. `lockCodes` attribute is updated with this information. Now user initiates scanning once again. Now DTH should not generate `$codeID changed` event because the code has not changed actually. The DTH will generate `$codeID set` event if a slot is found occupied during scanning and the information about the same is not available in `lockCodes` attribute. SmartApp can read the lock codes related information from `lockCodes` attribute. Also we have another attribute `scanCodes` whose value will change from scanning to complete. SmartApps can subscribe to `scanCodes` and read updated `lockCodes` attribute after scanning.

Reply:
Correct, see above